xc/device.ml

```diff
@@ -1819,19 +1820,29 @@ module Dm_Common = struct
     ; fd_map = []
     }
 
-  let vgpu_args_of_nvidia domid vcpus (vgpu:Xenops_interface.Vgpu.nvidia) pci restore device =
+  let vgpu_args_of_nvidia domid vcpus vgpus restore =
     let open Xenops_interface.Vgpu in
+    let virtual_pci_address_compare vgpu1 vgpu2 =
+      match vgpu1, vgpu2 with
+      |{implementation = Nvidia {virtual_pci_address = Some pci1; _}}, {implementation = Nvidia {virtual_pci_address = Some pci2; _}} -> Pervasives.compare pci1.dev pci2.dev
+      | _, _ -> raise (Xenopsd_error (Internal_error "Not comparable vGPUs"))
+      in
+    let get_pci_string = function
+      | None -> raise (Xenopsd_error (Internal_error "No PCI address"))
```

Comment:
Use `Printf.sprintf "No PCI address (%s)" __LOC__` This will make it easier to find this once it happens. Also below.